ui/segmented-bar.reel/segmented-bar.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * @module montage/ui/segmented-bar.reel
+ * @requires montage/ui/base/abstract-button.AbstractButton
+ * @requires montage/core/tree-controller.TreeController
+ */
+var AbstractButton = require("ui/base/abstract-button").AbstractButton;
+
+/**
+ * @class SegmentedBar
+ * @classdesc
+ * A segmented bar.
+ * Reference design: OS X storage space visualization found in iTunes or About This Mac
+ * Inheriting from AbstractButton because SegmentedBar needs its PressComposer behavior
+ *
+ * @extends AbstractButton
+ *
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Flexible_boxes}
+ * for FlexBox usage
+ */
+exports.SegmentedBar = AbstractButton.specialize( /** @lends SegmentedBar.prototype # */ {
```

Comment:
> Why is _treeControllerClass set lazily? Is this for performance? In any case a comment about this where _treeControllerClass is defined would help.

`require()` is done lazily for performance; I was doing 2 instance checks in 2 different places, hence the private property, but not anymore, will remove